core/src/main/java/io/stargate/core/metrics/StargateTenantExtractor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.core.metrics;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StargateTenantExtractor {
```

Comment:
I see that this is covering extraction of the tenant from headers, but not `ClientState` (for `CQL`) 
Do we plan to support it as well?

Reply:
Yes this will be supported as the next step, but will have this in mind when designing the interfaces/services..

Reply:
@tomekl007  Can this be resolved?

Reply:
Where is this implemented? Will it be in `c2`?

Reply:
Yes, my next task..